### PR TITLE
Make usage of Docker for development easier by avoiding COPY operations

### PR DESCRIPTION
**Pull request by paroga**

### CODE REVIEW

File: Dockerfile-dev [hunk 1]
```diff
@@ -25,15 +25,6 @@ USER app
 
 WORKDIR /home/app/src
 
-# Copy files needed for installing gem dependencies, and install them.
-COPY Gemfile Gemfile.lock ./
-COPY plugins ./plugins
-RUN bundle config build.nokogiri "--use-system-libraries" && \
```
Comment (wvengen) [inline Dockerfile-dev:31]: does `--use-system-libraries` still have any use, e.g. not having to compile libxml and friends at `bundle install` time?

File: doc/SETUP_DEVELOPMENT_DOCKER.md [hunk 2]
```diff
@@ -28,19 +28,20 @@ afterwards.
 
 ## Setup
 
-Then setup foodsoft development (this will take some time, containers needs
-to be pulled from docker registry and a lot dependencies needs to be installed)
+Then start the databasee server and setup foodsoft development (this will take
```
Comment (wvengen) [inline doc/SETUP_DEVELOPMENT_DOCKER.md:31]: small typo

File: doc/SETUP_DEVELOPMENT_DOCKER.md [hunk 3]
```diff
@@ -28,19 +28,20 @@ afterwards.
...
+    docker-compose -f docker-compose-dev.yml run --rm foodsoft \
+      bundle exec rake foodsoft:setup_development_docker
 
-Then create an initial database (here seeded with `small.en`) by running
+Optionally an initial database (here seeded with `small.en`) can be loaded by running
 
-    docker-compose -f docker-compose-dev.yml up -d mariadb && sleep 15
     docker-compose -f docker-compose-dev.yml run --rm foodsoft \
-      bundle exec rake db:create db:schema:load db:seed:small.en
+      bundle exec rake db:schema:load db:seed:small.en
```
Comment (wvengen) [inline doc/SETUP_DEVELOPMENT_DOCKER.md:44]: Let's be consistent in whether to use `\` or not (here and elsewhere in the file).

File: lib/tasks/foodsoft_setup.rake [hunk 4]
```diff
@@ -34,10 +34,20 @@ namespace :foodsoft do
     setup_development
     setup_database
     start_mailcatcher
-    puts yellow "All done! Your foodcoft should be running smoothly."
+    puts yellow "All done! Your foodsoft should be running smoothly."
```
Comment (wvengen) [inline lib/tasks/foodsoft_setup.rake:37]: What about "Your Foodsoft installation should be running smoothly." ? (or s/installation/setup/)

**Review by wvengen — COMMENTED** [review]
Great to make this better.

File: doc/SETUP_DEVELOPMENT_DOCKER.md (hunk 3 above)
Comment (paroga) [reply doc/SETUP_DEVELOPMENT_DOCKER.md:44]: IMHO it's consistent. It's only missing when there are multiple commands (vs. multiline commands)

Comment (wvengen) [reply doc/SETUP_DEVELOPMENT_DOCKER.md:44]: Yes, I see now that later invocations with `docker-compose-dev` don't use it - which is good.

**Comment by paroga** [comment]
I removed the user completely, since it works only if the uid `1000` matches (usually the first user in the system)...

**Comment by wvengen** [comment]
That's right, about the uid.
I'm a little hesitant about running Foodsoft as `root` by default, though (also to be more similar to production).
Could we avoid that?

**Comment by paroga** [comment]
as long as we can't assure that the uid in and outside the container matches this brings a bunch of permission issues. I agree with not running as root for production, but for development in docker? I found no better EASY solution for it, but feel free to bring up something better.

**Comment by wvengen** [comment]
Ok, maybe we can allow `root` for development. Let's see if it puts new developers off or not ;)
The only option would be to add all relevant files and dirs in Foodsoft's root directory as volumes - not pretty.

**Review by wvengen — APPROVED** [review]
Let's do it! :-)

**Pull request merged**